src/workloads/execution/SSBColumnStoreIndex.yml

```diff
@@ -0,0 +1,791 @@
+          }
+        }
+      },
+      {
+        "$lookup": {
+          "from": "supplier",
+          "localField": "l_suppkey",
+          "foreignField": "s_suppkey",
+          "as": "supplier"
+        }
+      },
+      {"$addFields": {"supplier": {"$first": "$supplier"}}},
+      {"$match": {"supplier.nation.n_name": "UNITED STATES"}},
+      {
+        "$lookup": {
+          "from": "customer",
+          "localField": "order.o_custkey",
+          "foreignField": "c_custkey",
+          "as": "customer"
+        }
+      },
+      {"$addFields": {"customer": {"$first": "$customer"}}},
+      {"$match": {"customer.nation.region.r_name": "AMERICA"}},
+      {
+        "$group": {
+          "_id": {
+            "order_year": {"$year": "$order.o_orderdate"},
+            "c_nation": "$customer.nation.n_name",
+            "p_brand": "$part.p_brand"
+          },
+          "profit": {
+            "$sum": {
+              "$subtract": [
+                "$l_extendedprice",
+                {"$multiply": ["$l_extendedprice", "$l_discount"]}
+              ]
+            }
+          }
+        }
+      },
+      {
+        "$addFields": {
+          "order_year": "$_id.order_year",
+          "customer_nation": "$_id.c_nation",
+          "part_brand": "$_id.p_brand"
+        }
+      },
+      {"$sort": {"order_year": 1, "customer_nation": 1, "part_brand": 1}}
+    ]
+
+# Defines how many times each test operation is executed. Using 3 due to variability in timing numbers.
+TestRepeatCount: &TestRepeatCount 3
+MaxPhases: &MaxPhases 42
+ColumnStoreIndexName: &ColumnStoreIndexName $**_columnstore
+
+Actors:
+- Name: Setup
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Thread: 1
+        Database: admin
+        Operations:
+        - OperationName: AdminCommand
+          OperationCommand:
+            setParameter: 1
+            internalQueryFrameworkControl: "trySbeEngine"
+
+- Name: BuildColumnStoreIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Thread: 1
+        Database: *db
+        Operations:
+        - OperationMetricsName: BulkBuildColumnStoreIndex
+          OperationName: RunCommand
+          OperationCommand:
+            createIndexes: *Collection
+            indexes:
+            - key: {"$**": "columnstore"}
+              name: *ColumnStoreIndexName
+
+# Quiesce before each query
+- Name: QuiescePhase
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+# Run each SSB query in a separate phase with column store index.
```

Comment:
Have you considered using an actor template? 

For larger workloads like this one, it will cut down on the size of the yml file, and more importantly it will mean less editing if you ever need to add / change something. For example in your case you would do something like:

```
# define before the actors.
ActorTemplates:
- TemplateName: SSB.Query
  Config:
    Name: {^Parameter: {Name: "Name", Default: "Q1_1"}}
    Type: CrudActor
    Phases:
      OnlyActiveInPhases:
        Active: {^Parameter: {Name: "Active", Default: [3]}}
        NopInPhasesUpTo: *MaxPhases
        PhaseConfig:
          Repeat: *TestRepeatCount
          Database: *db
          Collection: *Collection
          Operations:
          - OperationMetricsName: {^Parameter: {Name: "Name", Default: "Q1_1"}}
            OperationName: aggregate
            OperationCommand: {^Parameter: {Name: "OperationCommand", Default: *Q1_1Query}}

# skip
.
.
.


Actors:
# skip
.
.
.
.

- ActorFromTemplate:
    TemplateName: SSB.Query
    TemplateParameters:
      Name: Q1_1

- ActorFromTemplate:
    TemplateName: SSB.Query
    TemplateParameters:
      Name: Q1_2
      OperationCommand: *Q1_2Query
      Active: [5]

- ActorFromTemplate:
    TemplateName: SSB.Query
    TemplateParameters:
      Name: Q1_3
      OperationCommand: *Q1_3Query
      Active: [7]

- ActorFromTemplate:
    TemplateName: SSB.Query
    TemplateParameters:
      Name: Q2_1
      OperationCommand: *Q2_1Query
      Active: [9]

# etc/


# run same queries without column store index (Collection Scan)
- ActorFromTemplate:
    TemplateName: SSB.Query
    TemplateParameters:
      Name: Q1_1CollScan
      OperationCommand: *Q1_1Query
      Active: [24]

# etc.
```

Note: the metrics names will end up being the Template Parameter name but I don't see that as terribly significant.

You can view the processed workload with 
```
./run-genny evaluate src/workloads/execution/SSBColumnStoreIndex.yml
```